Library/Homebrew/test/dev-cmd/livecheck_spec.rb

```diff
@@ -20,7 +20,27 @@
       .and be_a_success
   end
 
-  it "gives an error when no arguments are given and there's no watchlist", :integration_test do
+  it "errors when a `livecheck` block URL symbol does not correspond to a checkable URL", :integration_test do
+    content = <<~RUBY
+      desc "Some test"
+      homepage "https://github.com/Homebrew/brew"
+      url "https://brew.sh/test-1.0.0.tgz"
+
+      # This is deliberately wrong, as `:url` should only be used in casks
+      # (the equivalent for formulae is `:stable`).
+      livecheck do
+        url :url
+      end
+    RUBY
+    setup_test_formula("test", content)
+
+    expect { brew "livecheck", "test" }
+      .to output(/`url :url` does not reference a checkable URL/).to_stderr
```

Comment:
```suggestion
      livecheck do
        url :badsymbol
      end
    RUBY
    setup_test_formula("test", content)

    expect { brew "livecheck", "test" }
      .to output(/`url :badsymbol` does not reference a checkable URL/).to_stderr
```
Seems more obvious, and avoids the need to add a comment?

Reply:
We could use a clearly invalid symbol but I opted for one that's valid but contextually incorrect, since that's an issue we've already encountered in practice. `:badsymbol` could error due to being invalid while a valid but contextually-incorrect symbol like `:stable`/`:url` may not error if we introduce a bug. Ideally we would test both scenarios but I wasn't sure if we wanted more than one test (with integration tests being slower), so I went with the more plausible scenario.

That said, it doesn't matter much either way if we improve test coverage of `#livecheck_url_to_string`. It seems to be lacking, so I'll see about adding that probably tonight or tomorrow.

Reply:
After some testing, it turns out that we have to use a valid symbol to be able to avoid the [`:badsymbol is not a valid URL shorthand` error](https://github.com/Homebrew/brew/blob/ddfe04a6013131d4f70a278360f4991011dca679/Library/Homebrew/livecheck.rb#L174) from the `#url` method. We want to be able to reach the error within livecheck, so it has to be a URL that's valid but doesn't correspond to a checkable URL for this formula and that leaves us with `:url`.